src/components/Select/Select.tsx

```diff
@@ -161,20 +161,20 @@ export const Select: FC<SelectProps> = React.forwardRef(
       168,
       8,
       1,
-      getSelectedOptionValues().length
+      getSelectedOptionValues()?.length
     );
 
     // Populate options on component render
     useEffect(() => {
-      const selected: SelectOption[] = options.filter(
-        (opt: SelectOption) => opt.selected
+      const selected: SelectOption[] = options?.filter(
+        (opt: SelectOption) => opt?.selected
       );
       setOptions(
-        _options.map((option: SelectOption, index: number) => ({
-          selected: !!selected.find((opt) => opt.value === option.value),
+        _options?.map((option: SelectOption, index: number) => ({
+          selected: !!selected?.find((opt) => opt?.value === option?.value),
           hideOption: false,
-          id: option.text + index,
-          object: option.object,
+          id: option?.text + '-' + index,
```

Comment:
perhaps loop functions can check that the type of option is correct and filter it out if not... things might go pretty sidewise if we are trying to support objects of the wrong type

Reply:
This sounds risky, shouldn't JSON validation be deferred to the host app?

Reply:
A few risks:

- Unnecessary parsing
- Duplicate validation impacting perf
- Runtime error if validation is missing some edge case